drivers/sensor/bh1749/bh1749.c

```diff
@@ -50,7 +50,7 @@ static int bh1749_sample_fetch(const struct device *dev, enum sensor_channel cha
 		return -ENOTSUP;
 	}
 
-	if (unlikely(!data->ready)) {
+	if (unlikely(!bh1749_wait_for_ready(dev))) {
```

Comment:
I find this condition a bit hard to read. Both because `unlikely` is not very common, and `wait_for_ready` is a bit misleading name - since it doesn't guarantee that it's ready when returning.

- The `unlikely` compiler optimization is probably harming readability more that it helps performance when this isn't in a performance critical loop. It can most probably be removed from the driver, but it's not related, so that can be cleaned up in a separate PR.
- The name `bh1749_wait_for_ready` implies that it will wait until it's ready, and it a bit hard to map the result of it to a boolean without reading the functions implementation. How `bh1749_safe_ready` or  just `bh1749_ready?

Reply:
Interesting, for me it's actually the other way aroud when I encounter `unlikely()`. It gives me some information about what the common case in the evaluation is, and is more a help than a negative.
I removed the `unlikely()` from the check in the various APIs, and kept it in the helper that will be called most frequently.

> The name bh1749_wait_for_ready implies that it will wait until it's ready, and it a bit hard to map the result of it to a boolean without reading the functions implementation. How bh1749_safe_ready or just `bh1749_ready?

Naming is the hardest part of programming, and this is no exception.. Changed it to `bh1749_is_ready()`

Reply:
> Naming is the hardest part of programming, and this is no exception.. Changed it to `bh1749_is_ready()`

I couldn't agree more! `bh1749_is_ready()` works for me :)